src/blog/2018-09-24/mutation-switching.pug

```diff
@@ -0,0 +1,116 @@
+extends ../../layout/blog-layout.pug
+block content
+    :markdown-it
+        
+        We are very excited because Stryker has a few new friends in the Stryker landscape! One of those new friends is Stryker4s(cala). 
+        With Stryker now being available for Scala, we opened up a whole new landscape for people to get started with mutation testing. 
+        The adaptation of Scala gave us a lot of joy but also a lot of new problems. One of those problems was the Scala compiler, 
+        we all know it's not the [fastest of its kind](https://scala-ci.typesafe.com/grafana/dashboard/db/scala-benchmark?from=1429139130109&to=1492531027936h&orgId=1&var-branch=2.11.x&var-branch=2.12.x&var-branch=2.13.x&var-source=scala&var-bench=HotScalacBenchmark.compile&var-host=scalabench@scalabench@) 
+        and many many hours are being put in to make it as fast as possible. With the goals of stryker in mind (It's easy to use and fast to run) 
+        we needed to come up with an intelligent way to deal with compilation time.
+        
+        ## Compiling each mutation
+        One way of introducing mutations to a codebase is mutating one statement, compile to code, run tests and gather the results. 
+        This is a logical choice because you have the maximum flexibility and are certain what effect the mutant had. 
+        When looking at the following example there are three mutations possible, we can change the `>` into `>=`, `<` and  `==`.
+        ```scala
+        class Numbers {
+            def greaterThan(a: Int, b: Int): Boolean = {
+                a > b
+            }
+        }
+        ```
+        If we apply the mutation one by one we would need to compile the code base three times. 
+        If we assume the compile time of this program is 10 seconds, we already have 30 seconds of compile time for one full mutation run. 
+        This would quickly get out of hand when the code base is bigger and has more mutations available.
+        
+        ## Mutation switching
+        Mutation switching to the rescue! So how is mutation switching faster than compiling each mutation on its own? 
+        The steps are quite similar but have some minor differences. First, All possible mutations are identified for the whole code base. 
+        Secondly, all mutations are applied to the code base using a [Scala Pattern match](https://docs.scala-lang.org/tour/pattern-matching.html) 
+        with the original satement in the default case, and activated using environment variables.
```

Comment:
typo: statement.

Also, this is a long sentence. Can we shorten it? Maybe move out the "and activated using environment variables" part to a new sentence?